core/src/main/java/com/sequenceiq/cloudbreak/service/SimpleSecurityService.java

```diff
@@ -0,0 +1,215 @@
+            Stack stack = stackRepository.findById(stackId);
+            if (stack != null) {
+                readServerCert(stack.getId());
+                readClientCert(stack.getId());
+                readClientKey(stack.getId());
+                readPrivateSshKey(stack.getId());
+                readPublicSshKey(stack.getId());
+            }
+    }
+
+    public String getSshPrivateFileLocation(Long stackId) {
+        return Paths.get(getCertDir(stackId) + "/" + getPrivateSshKeyFileName(stackId)).toString();
+    }
+
+    private String readSecurityFile(Long stackId, String fileName) throws CloudbreakSecuritySetupException {
+        try {
+            return FileReaderUtils.readFileFromPathToString(Paths.get(getCertDir(stackId) + "/" + fileName).toString());
+        } catch (IOException | SecurityException se) {
+            throw new CloudbreakSecuritySetupException("Failed to read file: " + getCertDir(stackId) + "/" + fileName);
+        }
+    }
+
+    private void writeSecurityFile(Long stackId, String content, String fileName) throws CloudbreakSecuritySetupException {
+        try {
+            String path = Paths.get(getCertDir(stackId) + "/" + fileName).toString();
+            File directory = new File(getCertDir(stackId));
+            if (!directory.exists()) {
+                Files.createDirectories(Paths.get(getCertDir(stackId)));
+            }
+            File file = new File(path);
+            if (!file.exists()) {
+                if (content != null) {
+                    FileOutputStream output = new FileOutputStream(file);
+                    IOUtils.write(Base64.decodeBase64(content), output);
+                }
+            }
+        } catch (IOException | SecurityException se) {
+            throw new CloudbreakSecuritySetupException("Failed to write file: " + getCertDir(stackId) + "/" + fileName);
+        }
+    }
+
+    private boolean checkSecurityFileExist(Long stackId, String fileName) throws CloudbreakSecuritySetupException {
+        try {
+            String path = Paths.get(getCertDir(stackId) + "/" + fileName).toString();
+            File directory = new File(getCertDir(stackId));
+            if (!directory.exists()) {
+                return false;
+            }
+            File file = new File(path);
+            if (!file.exists()) {
+                return false;
+            }
+        } catch (SecurityException se) {
+            throw new CloudbreakSecuritySetupException("Failed to check file: " + getCertDir(stackId) + "/" + fileName);
+        }
+        return true;
+    }
+
+    public void copyClientKeys(Path stackCertDir) throws IOException {
+        File file = new File(stackCertDir.toString());
+        if (!file.exists()) {
+            Files.createDirectories(stackCertDir);
+        }
+        Files.copy(Paths.get(clientPrivateKey), Paths.get(stackCertDir + "/key.pem"), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
It will be auto-generated by cbd (https://github.com/sequenceiq/cloudbreak-deployer/pull/80). But I'll implement some kind of pre-check that will validate the files when Cloudbreak starts.